pkg/rbac/sharedData.go

```diff
@@ -51,36 +55,103 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
+	if sqlerr != nil {
+		klog.Errorf("Error building Search query: %s", sqlerr.Error())
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	resourceTypeMap := make(map[string]string)
+	for rows.Next() {
+		var key, value string
+		err = rows.Scan(&key, &value)
+
+		if err != nil {
+			klog.Errorf("Error %s retrieving rows for query:%s", err.Error(), query)
+			continue
+		}
+		resourceTypeMap[key] = value
+
+	}
+	//cache query:
+	shared.propTypes = resourceTypeMap
+	shared.propTypeErr = err
+	shared.propTypeTime = time.Now()
+
+	return resourceTypeMap, err
+}
+
+func (cache *Cache) GetPropertyTypes(ctx context.Context) (map[string]string, error) {
+	sharedData, err := cache.PopulateSharedCache(ctx)
+	if err != nil {
+		klog.Error("Error populating shared data cache: ", err)
+		return nil, err
+	}
+	//store only the PropTypeCache to use in outside of rbac module
+	propTypesMap := sharedData.propTypes
+
+	return propTypesMap, nil
+}
+
+func (cache *Cache) PopulateSharedCache(ctx context.Context) (*SharedData, error) {
 
 	if sharedCacheValid(&cache.shared) { //if all cache is valid we use cache data
 		klog.V(5).Info("Using shared data from cache.")
-		return nil
+		return &cache.shared, nil
 	} else { //get data and cache
 
 		var error error
+
+		_, err := cache.shared.getPropertyTypes(cache, ctx)
+		if err != nil {
+			klog.Errorf("Error retrieving property resources. Error: [%+v]", err)
+		} else {
+			klog.V(6).Info("Successfully retrieved property resources!")
+		}
 		// get all cluster-scoped resources and cache in shared.csResources
-		err := cache.shared.GetClusterScopedResources(cache, ctx)
-		if err == nil {
-			klog.V(6).Info("Successfully retrieved cluster scoped resources!")
+		err = cache.shared.GetClusterScopedResources(cache, ctx)
+		if err != nil {
+			klog.Errorf("Error retrieving cluster scoped resources. Error: [%+v]", err)
 		} else {
-			error = err
+			klog.V(6).Info("Successfully retrieved cluster scoped resources!")
 		}
 		// get all namespaces in cluster and cache in shared.namespaces.
 		err = cache.shared.GetSharedNamespaces(cache, ctx)
-		if err == nil {
-			klog.V(6).Info("Successfully retrieved shared namespaces!")
+		if err != nil {
+			klog.Errorf("Error retrieving shared namespaces. Error: [%+v]", err)
 		} else {
-			error = err
+			klog.V(6).Info("Successfully retrieved shared namespaces!")
 		}
 		// get all managed clustsers in cache
 		err = cache.shared.GetManagedClusters(cache, ctx)
 		if err == nil {
-			klog.V(6).Info("Successfully retrieved managed clusters!")
+			klog.Errorf("Error retrieving managed clusters. Error: [%+v]", err)
 		} else {
-			error = err
+			klog.V(6).Info("Successfully retrieved managed clusters!")
 		}
-		return error
+
+		return &cache.shared, error
```

Comment:
Do shared cache need to be returned here? It is part of the cache and can be accessed from there.

Reply:
we can call on cache.

Reply:
Ok.  remove the &cache.shared from the return.